### PR TITLE
[FEMDEM] correcting some compilation errors in linux

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
In this PR I'm correcting & cleaning some errors in linux

### CODE REVIEW

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp [hunk 1]
```diff
@@ -39,6 +39,72 @@ void ExtendPressureConditionProcess<TDim>::GetMaximumConditionIdOnSubmodelPart(
...
+{
+    auto& r_geom = (*itElem)->GetGeometry();
+    std::vector<IndexType> condition_nodes_id(3);
+    condition_nodes_id[0] = r_geom[Id1].Id();
+    condition_nodes_id[1] = r_geom[Id2].Id();
+    condition_nodes_id[2] = r_geom[Id3].Id();
+    rMaximumConditionId++;
+
+    // Adding the nodes to the SubModelPart
+	rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id1].Id()));
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: I am quite sure that this is a Golberg machine. If the node already exists on the root modelpart it is enough with providing the Id, you are accessing to the database to the add it again  

https://en.wikipedia.org/wiki/Rube_Goldberg_machine

imagen

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp [hunk 2]
```diff
@@ -39,6 +39,72 @@ void ExtendPressureConditionProcess<TDim>::GetMaximumConditionIdOnSubmodelPart(
...
+    int& rMaximumConditionId
+    )
+{
+    std::vector<IndexType> condition_nodes_id(2);
+    auto& r_geom = (*itElem)->GetGeometry();
+    condition_nodes_id[0] = r_geom[Id1].Id();
+    condition_nodes_id[1] = r_geom[Id2].Id();
+    rMaximumConditionId++;
+
+    rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id1].Id()));
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: Idem

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp [hunk 3]
```diff
@@ -39,6 +39,72 @@ void ExtendPressureConditionProcess<TDim>::GetMaximumConditionIdOnSubmodelPart(
...
+    rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id2].Id()));
+
+    const auto& r_line_cond = rSubModelPart.CreateNewCondition(
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: This is a pointer, p_line_cond, and you don't need the &

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Some minor things.anyway this looks more like a refactor than a compilation fix on Linux

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: Im accessing the model part for adding that node to one SubModel, it is not the same I think :)

Comment (loumalouomega) [reply applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: The node is defined on the root model part, try to use the Id, is faster

**Comment by AlejandroCornejo** [comment]
> Some minor things.anyway this looks more like a refactor than a compilation fix on Linux

The major error was related to the order of instantiation os the templates in that file... This is wy it seems to be a refactor

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp [hunk 4]
```diff
@@ -60,19 +60,17 @@ void ExtendPressureConditionProcess<3>::CreatePressureLoads(
     rMaximumConditionId++;
 
     // Adding the nodes to the SubModelPart
-	rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id1].Id()));
-	rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id2].Id()));
-	rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id3].Id()));
+	rSubModelPart.AddNodes(condition_nodes_id);
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp:63]: :+1:

**Review by loumalouomega — APPROVED** [review]
OK

**Pull request merged**